### PR TITLE
TST: address tolerance violations with SciPy 1.14.0rc1 on linux-{aarch64,ppc64le}

### LINKED ISSUE #20938: TST: tolerance violations with SciPy 1.14.0rc1 on linux-{aarch64,ppc64le}
In https://github.com/conda-forge/scipy-feedstock/pull/277, I see new failures for aarch/ppc.

On linux-aarch64, there are some minor violations of relative tolerance:
```
=========================== short test summary info ============================
FAILED special/tests/test_hyp2f1.py::TestHyp2f1::test_region2[hyp2f1_test_case15] - AssertionError: 
Not equal to tolerance rtol=1e-15, atol=0

Mismatched elements: 1 / 1 (100%)
Max absolute difference among violations: 7.42004997e-11
Max relative difference among violations: 1.24886275e-15
 ACTUAL: array(54749.216391-23078.144721j)
 DESIRED: array(54749.216391-23078.144721j)
FAILED special/tests/test_hyp2f1.py::TestHyp2f1::test_region5[hyp2f1_test_case44] - AssertionError: 
Not equal to tolerance rtol=1.1e-15, atol=0

Mismatched elements: 1 / 1 (100%)
Max absolute difference among violations: 1.24126708e-15
Max relative difference among violations: 1.51901296e-15
 ACTUAL: array(-0.185152+0.795901j)
 DESIRED: array(-0.185152+0.795901j)
= 2 failed, 49224 passed, 2743 skipped, 159 xfailed, 14 xpassed, 106 warnings in 6336.31s (1:45:36) =
```

On linux-ppc64le, the failures are in completely different tests; while also related to tolerance violations, here it looks more like it's about values close to machine precision, and thus the absolute tolerance that would need to be loosened.

``` 
=========================== short test summary info ============================
FAILED optimize/tests/test_differentiate.py::TestDifferentiate::test_vectorization[shape1-6] - AssertionError: 
Not equal to tolerance rtol=1e-07, atol=5e-15

Mismatched elements: 1 / 12 (8.33%)
Max absolute difference among violations: 6.02295991e-15
Max relative difference among violations: 2.00249219e-06
 ACTUAL: array([4.384204e-09, 4.384201e-09, 4.081055e-09, 3.501619e-09,
       2.696714e-09, 1.735705e-09, 6.991865e-10, 3.291873e-10,
       1.270939e-09, 2.059971e-09, 2.647979e-09, 3.007738e-09])
 DESIRED: array([4.384204e-09, 4.384203e-09, 4.081054e-09, 3.501620e-09,
       2.696713e-09, 1.735705e-09, 6.991875e-10, 3.291881e-10,
       1.270939e-09, 2.059969e-09, 2.647979e-09, 3.007732e-09])
FAILED optimize/tests/test_differentiate.py::TestDifferentiate::test_vectorization[shape2-6] - AssertionError: 
Not equal to tolerance rtol=1e-07, atol=5e-15

Mismatched elements: 1 / 12 (8.33%)
Max absolute difference among violations: 6.02295991e-15
Max relative difference among violations: 2.00249219e-06
 ACTUAL: array([4.384204e-09, 4.384201e-09, 4.081055e-09, 3.501619e-09,
       2.696714e-09, 1.735705e-09, 6.991865e-10, 3.291873e-10,
       1.270939e-09, 2.059971e-09, 2.647979e-09, 3.007738e-09])
 DESIRED: array([4.384204e-09, 4.384203e-09, 4.081054e-09, 3.501620e-09,
       2.696713e-09, 1.735705e-09, 6.991875e-10, 3.291881e-10,
       1.270939e-09, 2.059969e-09, 2.647979e-09, 3.007732e-09])
FAILED optimize/tests/test_differentiate.py::TestDifferentiate::test_vectorization[shape3-6] - AssertionError:
Not equal to tolerance rtol=1e-07, atol=5e-15

Mismatched elements: 1 / 12 (8.33%)
Max absolute difference among violations: 6.02295991e-15
Max relative difference among violations: 2.00249219e-06
 ACTUAL: array([4.384204e-09, 4.384201e-09, 4.081055e-09, 3.501619e-09,
       2.696714e-09, 1.735705e-09, 6.991865e-10, 3.291873e-10,
       1.270939e-09, 2.059971e-09, 2.647979e-09, 3.007738e-09])
 DESIRED: array([4.384204e-09, 4.384203e-09, 4.081054e-09, 3.501620e-09,
       2.696713e-09, 1.735705e-09, 6.991875e-10, 3.291881e-10,
       1.270939e-09, 2.059969e-09, 2.647979e-09, 3.007732e-09])
= 3 failed, 49562 passed, 2741 skipped, 160 xfailed, 16 xpassed, 106 warnings in 5716.31s (1:35:16) =
```

In both cases it's possible that this is related to QEMU version and/or details of the underlying host agent. I can skip those tests if necessary, but it would be nicer to fix it properly.

### PR DESCRIPTION
**Pull request by mdhaber**
#### Reference issue
Closes gh-20938

#### What does this implement/fix?
Bumps tolerances to address local test failures reported in gh-20938.

#### Additional information
The `test_differentiate` adjustment is already needed in another PR and seems safe to me (the author). The `hyp2f1` adjustments are very small, but I'll let @steppi check that they're acceptable.

@h-vetinari Can you check that this addresses the failures?

### CODE REVIEW

**Comment by steppi** [comment]
The tolerance bumps for `hyp2f1` look fine to me.

File: scipy/optimize/tests/test_differentiate.py [hunk 1]
```diff
@@ -64,7 +64,7 @@ def f(x, *args, **kwargs):
         assert_equal(res.df.shape, shape)
 
         ref_error = [ref.error for ref in refs]
-        assert_allclose(res.error.ravel(), ref_error, atol=5e-15)
+        assert_allclose(res.error.ravel(), ref_error, atol=1e-12)
```
Comment (h-vetinari) [inline scipy/optimize/tests/test_differentiate.py:67]: I think we can be less aggressive here
```suggestion
        assert_allclose(res.error.ravel(), ref_error, atol=1e-14)
```

**Review by h-vetinari — APPROVED** [review]
This passed in conda-forge with the smaller atol bump (i.e. 1e-14 in `test_differentiate.py`)

I suggest to adopt this here, but otherwise LGTM - thank you!

**Comment by mdhaber** [comment]
I bumped to 1e-12 because I think that was needed with another array library (maybe PyTorch) in my other PR. Probably some smaller difference was causing it to not detect convergence and an additional iteration makes the absolute error a little worse. So it will become 1e-12 shortly if not here.

**Review by tylerjereddy — APPROVED** [review]
Two happy core devs, sounds like it helps `conda-forge`  situation, and our CI is green. Let's put it in for the next RC, thanks all.

**Pull request merged**